src/main/java/com/metamx/emitter/core/HttpPostEmitter.java

```diff
@@ -280,8 +283,11 @@ public void run()
           for (final List<byte[]> batch : batches) {
             log.debug("Sending batch to url[%s], batch.size[%,d]", url, batch.size());
 
+            byte[] decodedBatch = config.getCompress() ? serializeAndCompressBatch(batch) : serializeBatch(batch);
```

Comment:
This method belongs to the private inner class EmittingRunnable. It's possible to test the method by changing class scope to protected. Is this way appropriate for us?